webdav/pkg/dav/requests/thumbnail.go

```diff
@@ -0,0 +1,95 @@
+package requests
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/go-chi/chi"
+)
+
+const (
+	// DefaultWidth defines the default width of a thumbnail
+	DefaultWidth = 32
+	// DefaultHeight defines the default height of a thumbnail
+	DefaultHeight = 32
+)
+
+// ThumbnailRequest combines all parameters provided when requesting a thumbnail
+type ThumbnailRequest struct {
+	Filepath        string
+	Filename        string
+	Extension       string
+	Width           int32
+	Height          int32
+	PublicLinkToken string
+}
+
+// ParseThumbnailRequest extracts all required parameters from a http request.
+func ParseThumbnailRequest(r *http.Request) (ThumbnailRequest, error) {
+	fp := extractFilePath(r)
+	q := r.URL.Query()
+
+	width, height, err := parseDimensions(q)
+	if err != nil {
+		return ThumbnailRequest{}, err
+	}
+
+	tr := ThumbnailRequest{
+		Filepath:        fp,
+		Filename:        filepath.Base(fp),
+		Extension:       filepath.Ext(fp),
+		Width:           int32(width),
+		Height:          int32(height),
+		PublicLinkToken: chi.URLParam(r, "token"),
+	}
+
+	return tr, nil
+}
+
+// the url looks as followed
+//
+// /remote.php/dav/files/<user>/<filepath>
+//
+// User and filepath are dynamic and filepath can contain slashes
+// So using the URLParam function is not possible.
+func extractFilePath(r *http.Request) string {
+	user := chi.URLParam(r, "user")
+	if user != "" {
+		parts := strings.SplitN(r.URL.Path, user, 2)
+		return parts[1]
+	}
+	token := chi.URLParam(r, "token")
+	if token != "" {
+		parts := strings.SplitN(r.URL.Path, token, 2)
+		return parts[1]
+	}
+	return ""
```

Comment:
Is returning an empty filePath valid? `/remote.php/dav/files/refs/` seems to be a different endpoint altogether. It should rather fail here if for whatever reason (i.e: a crafted URL to the thumbnails service) a request made it to this point 🤷